### PR TITLE
Refactor viewmodels

### PR DESCRIPTION
**Pull request by codergrl**
@nCastle1 please review. This PR refactors duplicate code from viewmodels into a FeatureViewModel and handles all the aftermath from that. If you test this, please only test with WPF as I have not rewired the UWP UI to account for the changes. This PR is large enough as it is, I figured I don't need to add more.

### CODE REVIEW

File: src/DataCollection.WPF/DataCollection.WPF.csproj [hunk 1]
```diff
@@ -9,13 +9,14 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Esri.ArcGISRuntime.ExampleApps.DataCollection.WPF</RootNamespace>
     <AssemblyName>DataCollection.WPF</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Comment (nCastle1) [inline src/DataCollection.WPF/DataCollection.WPF.csproj]: Was this change intentional?

File: src/DataCollection.Shared/ViewModels/OriginRelationshipViewModel.cs [hunk 2]
```diff
@@ -232,132 +79,11 @@ public ICommand SaveEditsCommand
...
                     {
-                        var feature = await EditViewModel.SaveEdits(PopupManager, RelatedTable, null);
-
-                        if (feature != null)
-                        {
-                            // if a new record was added, we want to refresh the list so it shows up in the UI
-                            GetAddedRecordAndRefresh(feature);
-                        }
-
+                        var feature = await EditViewModel.SaveEdits(PopupManager, FeatureTable, null);
```
Comment (nCastle1) [inline src/DataCollection.Shared/ViewModels/OriginRelationshipViewModel.cs]: It looks like this variable is never used, is that intentional?

File: src/DataCollection.WPF/Views/IdentifiedFeaturePopup.xaml [hunk 3]
```diff
@@ -149,16 +149,18 @@
                                                     <TextBlock Text="{Binding DisplayedFields[2].FormattedValue}" FontSize="12" Foreground="Gray" Margin="7,0,0,10"/>
                                                 </StackPanel>
                                                 <Button Content="&#xE76C;" FontFamily="Segoe MDL2 Assets" FontSize="15" Width="30" Height="30" VerticalAlignment="Center" Style="{StaticResource MenuButtonStyle}" 
-                                                        Grid.Column="1" Grid.Row="1" HorizontalAlignment="Right" Margin="0,0,5,0" 
-                                                        ToolTip="{Binding ., Converter={StaticResource LocalizationConverter}, ConverterParameter=ShowDetailButton_Tooltip}"
-                                                        Command="{Binding DataContext.IdentifiedFeatureViewModel.SetSelectedOriginRelationshipCommand, RelativeSource={RelativeSource AncestorType=UserControl}}"
-                                                        CommandParameter="{Binding}">
+                                                    Grid.Column="1" Grid.Row="1" HorizontalAlignment="Right" Margin="0,0,5,0"
```
Comment (nCastle1) [inline src/DataCollection.WPF/Views/IdentifiedFeaturePopup.xaml]: The XAML files are starting to get a little unwieldy, especially with line wrapping turned on. Not for this PR, but it would be worthwhile doing an automated reformatting to make the XAML easier to read in VS.

**Review by nCastle1 — APPROVED** [review]
Changes look great! A few minor comments, feel free to take or leave. I tried the WPF version extensively and wasn't able to break it.

**Comment by nCastle1** [comment]
One more note: I noticed that the readme refers to 100.4, since this updates Runtime to 100.5, the readme should also be updated.

File: src/DataCollection.WPF/DataCollection.WPF.csproj (hunk 1 above)
Comment (codergrl) [reply src/DataCollection.WPF/DataCollection.WPF.csproj]: Kinda, I was trying something out that I ended up not using. I'll revert.

File: src/DataCollection.Shared/ViewModels/OriginRelationshipViewModel.cs (hunk 2 above)
Comment (codergrl) [reply src/DataCollection.Shared/ViewModels/OriginRelationshipViewModel.cs]: Changed it to save the value in the Feature property

File: src/DataCollection.WPF/Views/IdentifiedFeaturePopup.xaml (hunk 3 above)
Comment (codergrl) [reply src/DataCollection.WPF/Views/IdentifiedFeaturePopup.xaml]: Oh, that's a great idea! I just did that and it looks so much better. Thanks for the suggestion!
